kamelet-support/src/main/java/io/kaoto/backend/metadata/parser/step/camelRoute/CamelRouteFileProcessor.java

```diff
@@ -0,0 +1,275 @@
+                parameterData.get("description"),
+                parameterData.getOrDefault("defaultValue", null),
+                null
+        );
+
+        return parsedParameter;
+    }
+
+    private NumberParameter getNumberParameter(
+            final Map.Entry<String, Map<String, String>> parameter) {
+        //IntegerParameter doesn't fit some long types there are present
+        String id = parameter.getKey();
+        Map<String, String> parameterData = parameter.getValue();
+
+        NumberParameter parsedParameter = new NumberParameter(
+                id,
+                parameterData.get("displayName"),
+                parameterData.get("description"),
+                Long.parseLong(parameterData
+                        .getOrDefault("defaultValue", "0")
+                        .replaceAll("L", ""))
+        );
+
+        return parsedParameter;
+    }
+
+    private ObjectParameter getObjectParameter(
+            final Map.Entry<String, Map<String, String>> parameter) {
+        String id = parameter.getKey();
+        Map<String, String> parameterData = parameter.getValue();
+
+        ObjectParameter parsedParameter = new ObjectParameter(
+                id,
+                parameterData.get("displayName"),
+                parameterData.get("description"),
+                parameterData.getOrDefault("defaultValue", null)
+        );
+
+        return parsedParameter;
+    }
+
+    private BooleanParameter getBooleanParameter(
+            final Map.Entry<String, Map<String, String>> parameter) {
+        String id = parameter.getKey();
+        Map<String, String> parameterData = parameter.getValue();
+
+        BooleanParameter parsedParameter = new BooleanParameter(
+                id,
+                parameterData.get("displayName"),
+                parameterData.get("description"),
+                Boolean.parseBoolean(parameterData
+                        .getOrDefault("defaultValue", "false"))
+        );
+
+        return parsedParameter;
+    }
+
+    private String getStepType(final JsonObject component) {
+        final String sourceType = "source";
+        final String sinkType = "sink";
+        final String actionType = "action";
+
+        final boolean canBeSource = component.getBoolean("producerOnly");
+        final boolean canBeSink = component.getBoolean("consumerOnly");
+
+        //If both are true or false we assume it to be an action.
+        //Or else, it has to be one of source or sink depending
+        //Take care that this later has to be duplicated because
+        //of the differences in how Camel components and kaoto components are
+        return canBeSource == canBeSink ? actionType
+                : canBeSource ? sourceType
+                : sinkType;
+    }
+
+    // Putting it at the end because is just a big string being
+    // used as placeholder
+    private final String defaultIconString = "data:image/svg+xml;base64"
```

Comment:
We need to address the issue that there is no icon image for any of the camel components. We might have to create a side repo or something, we will have to see